src/components/menu/menu.svelte

```diff
@@ -0,0 +1,234 @@
+<script context="module" lang="ts">
+  import type { HTMLAttributes } from 'svelte/elements'
+  declare global {
+    namespace JSX {
+      interface IntrinsicElements {
+        'leo-menu-item': HTMLAttributes<HTMLElement> & {
+          // Note: This should line up with Reacts key type, but we don't want
+          // to depend on React in this layer, so we just define it manually.
+          key?: string | number | null
+          children?: any
+        }
+      }
+    }
+  }
+</script>
+
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte'
+  import clickOutside from '../../svelteDirectives/clickOutside'
+  import Floating from '../floating/floating.svelte'
+
+  export let isOpen = false
+  export let target: HTMLElement | undefined = undefined
+  export let currentValue: string | undefined = undefined
+
+  let dispatch = createEventDispatcher()
+
+  let popup: HTMLDivElement
+
+  function getValue(e: Element) {
+    // If the option element doesn't have a value, fallback to using the text
+    // content - this allows writing simplified options:
+    // i.e. <o>1</o>
+    return e.getAttribute('value') ?? e.textContent
+  }
+
+  $: menuItems = Array.from(
+    (
+      popup?.querySelector('.leo-menu-popup slot') as HTMLSlotElement
+    )?.assignedElements() ??
+      popup?.querySelectorAll('.leo-menu-popup > *') ??
+      []
+  )
+
+  $: {
+    for (const [menuItem, index] of menuItems.map((o, i) => [o, i] as const)) {
+      menuItem.setAttribute('tabindex', (index + 1).toString())
+
+      if (menuItem.tagName === 'LEO-OPTION') {
```

Comment:
This probably needs to be `LEO-MENU-ITEM` now

Reply:
Optionally this could be a prop, or we could come up with some cleverer way of doing this

Reply:
Why LEO-MENU-ITEM? aria-selected is set for LEO-OPTION only, right?

Reply:
Okay, sorry, I misunderstood how this was working - I assumed it was a holdover from the dropdown. Maybe instead of handling this via the type of the children, we could add a role prop to the `Menu` component, and do it based on that.

I worry that doing it based on the type of the children might be a bit non-obvious. WDYT @petemill?

Reply:
It would be simpler for the users to simply only have leo-menu-item, with a role prop for the container to check for these conditional aria attrubutes.

One minor consideration would be menus whose popups behave more like lists, e.g. a "..." menu for choosing an AI model. It's basically a dropdown without the input. Should we allow that mechanism via choosing <leo-option> vs <leo-menu-item>? Or should it be via the container? What about a menu that has a mix of selectable options and menu-items? Will our devs be aware of this enough?

Happy to merge it like this if you think it's the way to go, though IMO let's specify in the Storybook when to use which. Also, are we missing `role="option"` for `leo-option` items?

Reply:
I have added role=option for leo-option items. 

Let's see if the difference between 'leo-option' and 'leo-menu-item' is obvious to consumers, and I believe the first consumer for this would be AIChat. I have added a note regarding this in Storybook for both ButtonMenu and Dropdown.